driver-async/src/main/com/mongodb/async/client/MongoClientSettings.java

```diff
@@ -74,46 +76,36 @@ public static Builder builder(final MongoClientSettings settings) {
         return new Builder(settings);
     }
 
+    static MongoClientSettings createFromClientSettings(final com.mongodb.MongoClientSettings wrapped) {
+        return new Builder(wrapped).build();
+    }
+
     /**
      * A builder for {@code MongoClientSettings} so that {@code MongoClientSettings} can be immutable, and to support easier construction
      * through chaining.
      */
     @NotThreadSafe
     public static final class Builder {
+        private final com.mongodb.MongoClientSettings.Builder wrappedBuilder;
         private List<MongoCredential> credentialList = Collections.emptyList();
-        private final com.mongodb.MongoClientSettings.Builder wrappedBuilder = com.mongodb.MongoClientSettings.builder();
         private SocketSettings.Builder heartbeatSocketSettingsBuilder = null;
 
         private Builder() {
+            wrappedBuilder = com.mongodb.MongoClientSettings.builder();
+        }
+
+        private Builder(final com.mongodb.MongoClientSettings settings) {
+            notNull("settings", settings);
+            wrappedBuilder = com.mongodb.MongoClientSettings.builder(settings);
```

Comment:
Looks like a credential in the wrapped MongoClientSettings will be ignored.  I think this.credentialList has to be initialized with the wrappedBuilder.credential if it's non-null.